builds/release/release-core-images.yaml

```diff
@@ -0,0 +1,540 @@
+pr: none
+trigger: none
+
+resources:
+  pipelines:
+  - pipeline: detect-image-updates
+    source: 'Detect Image Updates'
+    trigger:
+      branches:
+        include:
+        - release/1.4
+  repositories:
+  - repository: product
+    type: github
+    endpoint: Azure/azure-iotedge
+    name: Azure/azure-iotedge
+
+variables:
+  DisableDockerDetector: true
+
+stages:
+################################################################################
+- stage: PrepareRelease
+################################################################################
+  displayName: Prepare for release
+  dependsOn: []
+
+  pool:
+    name: $(pool.linux.name)
+    demands:
+    - ImageOverride -equals agent-aziotedge-ubuntu-20.04-docker
+
+  jobs:
+  - job: PipelineTrigger
+    displayName: Trigger on base image updates
+    condition: eq(variables['Build.Reason'], 'ResourceTrigger')
+
+    steps:
+    - checkout: self
+      fetchDepth: 0
+
+    - download: detect-image-updates
+      displayName: Download image list from triggering pipeline
+
+    - script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        echo 'Updates:'
+        cat $(Pipeline.Workspace)/detect-image-updates/image-updates/updates.json
+
+        . scripts/linux/release-tools.sh
+        previous_version=$(get_nearest_version)
+        product='aziot-edge'
+        echo "Filtering on { product: $product, version: $previous_version }..."
+
+        # The images we build from this pipeline all target .NET Alpine base images, but some are
```

Comment:
Why does this imply the release may happen twice? It has something to do with dotnet aspnet vs dotnet runtime, but I don't understand still.

Reply:
We use two different .NET base images: temp sensor and diagnostics use dotnet/runtime, while Edge Agent and Edge Hub use dotnet/aspnet. Assuming that those two images (or even platform variations of the same image) aren't published to MCR at _exactly_ the same time, then it's possible that our pipeline will wake up and compare digests while the .NET team is in the middle of updating their images.

As an example, let's say the .NET is in the middle of a release, and their `dotnet/runtime` images have been published to MCR, but their `dotnet/aspnet` images have not (yet). Here's what could happen:
1. Detect-image-updates wakes up, detects that temp sensor and diagnostics images need to be refreshed because their base images have changed
2. Release-core-images is triggered and releases _all_ images, not just temp sensor and diagnostics
4. 12 hours later, Detect-image-updates wakes up and detects that Edge Agent and Edge Hub images need to be refreshed
5. Release-core-images is triggered and releases _all_ images, not just Edge Agent and Edge Hub.

In this case, we'd have two releases (two PATCH version bumps) in less than 24 hours, where one release would have sufficed.